tests/scribeAuth.test.ts

```diff
@@ -2,101 +2,211 @@ import { Auth, Tokens } from '@scribelabsai/auth';
   userPoolId: userPoolId,
   identityPoolId: federatedPoolId,
 });
 
+const mfaActivatedError = 'challenge instead of tokens';
+
 describe('Get tokens', () => {
   it('Username and password passes', async () => {
     const tokens = await access.getTokens({ username, password });
-    expect(assertTokens(tokens)).toBeTruthy();
+    if ('refreshToken' in tokens && 'idToken' in tokens && 'accessToken' in tokens) {
+      expect(assertTokens(tokens)).toBeTruthy();
+    } else {
+      throw new Error(mfaActivatedError);
+    }
   });
   it('Wrong username fails', async () => {
-    await expect(() => access.getTokens({ username: 'username', password })).rejects.toThrowError();
+    await expect(() => access.getTokens({ username: 'username', password })).rejects.toThrow();
   });
   it('Wrong password fails', async () => {
-    await expect(() => access.getTokens({ username, password: 'password' })).rejects.toThrowError();
+    await expect(() => access.getTokens({ username, password: 'password' })).rejects.toThrow();
   });
 
   it('Empty username fails', async () => {
-    await expect(() => access.getTokens({ username: '', password })).rejects.toThrowError();
+    await expect(() => access.getTokens({ username: '', password })).rejects.toThrow();
   });
   it('Empty password fails', async () => {
-    await expect(() => access.getTokens({ username, password: '' })).rejects.toThrowError();
+    await expect(() => access.getTokens({ username, password: '' })).rejects.toThrow();
   });
   it('Empty username and password fails', async () => {
-    await expect(() => access.getTokens({ username: '', password: '' })).rejects.toThrowError();
+    await expect(() => access.getTokens({ username: '', password: '' })).rejects.toThrow();
   });
   it('RefreshToken passes', async () => {
     const refreshToken = await getRefreshToken();
     const tokens = await access.getTokens({ refreshToken });
-    expect(assertTokens(tokens)).toBeTruthy();
+    if ('refreshToken' in tokens && 'idToken' in tokens && 'accessToken' in tokens) {
+      expect(assertTokens(tokens)).toBeTruthy();
+    } else {
+      throw new Error(mfaActivatedError);
+    }
   });
   it('Wrong refreshToken fails', async () => {
-    await expect(() => access.getTokens({ refreshToken: 'refresh_token' })).rejects.toThrowError();
+    await expect(() => access.getTokens({ refreshToken: 'refresh_token' })).rejects.toThrow();
   });
 });
 
 describe('Federated Credentials', () => {
   it('Get federated id passes', async () => {
     const tokens = await poolAccess.getTokens({ username, password });
-    const idToken = tokens.idToken;
-    const federatedId = await poolAccess.getFederatedId(idToken);
-    expect(federatedId).toBeDefined();
+    if ('idToken' in tokens) {
+      const idToken = tokens.idToken;
+      const federatedId = await poolAccess.getFederatedId(idToken);
+      expect(federatedId).toBeDefined();
+    } else {
+      throw new Error(mfaActivatedError);
```

Comment:
why are we throwing from inside the test?

Reply:
same comment as above